### ISSUE TITLE
catch.hpp and offline builds....

### ISSUE THREAD

**Issue by dchapiesky** [standardese/standardese#175]
Mirroring cppast's issues...  https://github.com/foonathan/cppast/issues/111

A gentle reminder that some users do solely offline builds.  While being connected to the internet to play with something is fine, there are occasions when:

A) you don't want scripts downloading stuff on your development servers
B) you want to reproduce a build exactly and downloading components each build defeats that

Not complaining but it makes it a burden to constantly merge modifications that fix these issues.

cppast and standardese look really really powerful, but all the file(DOWNLOAD ...)'s and forced git submodule update --init's really stink.

Simply allowing a recursive submodule update at the time of cloning and removing the forced execute_process() calls would really make it easier to add standardese and cppast to projects that are also designed to be built offline.

Sorry for the complaining.   Otherwise both projects are really really good.

**Comment by saraedum** [comment]
I agree with all you write. In the case of catch, it could as well be just another submodule instead of a download. I am not a fan of the download magic either and prefer to have a defined set of prerequisites instead (and scripts that can be run manually to establish them if there is a standard way to do so.)

So, from my end, any PRs to improve this are very welcome.

**Comment by dchapiesky** [comment]
I will see what I can put together...   
cheers
